File: php/features.php
<?php
declare(strict_types=1);

// EmailVerification SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EmailVerificationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EmailVerificationBaseFeature();
            case "test":
                return new EmailVerificationTestFeature();
            default:
                return new EmailVerificationBaseFeature();
        }
    }
}

File: py/emailverification_sdk/features.py
# EmailVerification SDK feature factory

from emailverification_sdk.feature.base_feature import EmailVerificationBaseFeature
from emailverification_sdk.feature.test_feature import EmailVerificationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EmailVerificationBaseFeature(),
        "test": lambda: EmailVerificationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
